### PR TITLE
Fixes agent IDs not working when you are put into a different body

### LINKED ISSUE #15986: Agent IDs and Contractor Pinpointers break after cloning
**Issue Description**:
Agent ID Cards and Contractor Pinpointers stop working if their owner has been cloned.

**What did you expect to happen**: 
The ID and Pinpointer would recognise you as their owner.

**What happened instead**:
No customisation option for the ID card ("You flash your ID card"), and the Pinpointer "Refuses to do anything".

**Why is this bad/What are the consequences**:
This is primarily an issue for the Agent ID Card, since those cost 2-4 TC and lose what is arguably its main functionality.

For the pinpointers, you get 2 reputation points for every contract you complete and they only cost 1, so it's really not that big of a problem. It still seems like unintended behaviour though so it might be worth fixing anyway.

**Steps to reproduce the problem**:
1. Link an Agent ID Card and/or Pinpointer to yourself.
2. Die and get cloned. (Or reincarnate)
3. Attempt to use either of the items.

**When did the problem start happening**:
Presumably when the items were added to the game.

**Extra information**:
The `registered_user` / `owner` variables are both linked to the actual mob, as opposed to the DNA or mind.
https://github.com/ParadiseSS13/Paradise/blob/58d7054e92a2a67adf1e78da8ad524fdba8f2504/code/game/objects/items/weapons/cards_ids.dm#L334
https://github.com/ParadiseSS13/Paradise/blob/58d7054e92a2a67adf1e78da8ad524fdba8f2504/code/modules/antagonists/traitor/contractor/items/contractor_pinpointer.dm#L14

### LINKED ISSUE #21756: Agent ID is tied to body over mind or dna.
**Issue Description**:

Agent ID gets tied to body, so if you are cloned you can no longer use it.

**What did you expect to happen**: 

for it to work off mind, or possibly dna

**What happened instead**:

cloned people can't use their agent ID

**Why is this bad/What are the consequences**:

Agent ids break if the user is cloned, which is funny.

It also means if somone is mindswapped by a cling, the agent id breaks, also funny.

**Steps to reproduce the problem**:

die
get cloned
try to use the agent ID you used earlier

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Ties the owner of the agent ID to the mind instead of the body, so if your brain gets transplanted or you get cloned, you can still edit the ID
Resolves #15986
Resolves #21756

## Why It's Good For The Game
Bug bad, I really don't have much to say here honestly.

## Testing
Started out as a human, assfaxed, got gibbed and put inside a grey.
I could still use my already activated agent ID afterwards.

## Changelog
:cl:
fix: Fixed agent IDs not working after being put into a different body
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -410,10 +410,10 @@
 		src.assignment = u
 		src.name = "[src.registered_name]'s ID Card ([src.assignment])"
 		to_chat(user, "<span class='notice'>You successfully forge the ID card.</span>")
-		registered_user = user
-	else if(!registered_user || registered_user == user)
+		registered_user = user.mind.current
```
Comment (SteelSlayer) [inline code/game/objects/items/weapons/cards_ids.dm:413]: Can you add a `registered_user = null` in the agent ID destroy proc also?

File: code/modules/antagonists/traitor/contractor/items/contractor_pinpointer.dm [hunk 2]
```diff
@@ -39,10 +39,10 @@
 
 /obj/item/pinpointer/crew/contractor/attack_self(mob/living/user)
 	if(owner)
-		if(owner != user)
+		if(owner != user.mind.current)
```
Comment (SteelSlayer) [inline code/modules/antagonists/traitor/contractor/items/contractor_pinpointer.dm:42]: Why did you change this code for the contractor pinpointer?

Comment (DGamerL) [reply code/modules/antagonists/traitor/contractor/items/contractor_pinpointer.dm:42]: The contractor pinpointer also should work after a cloning or brain transplant, as that was in the issue report too, unless I'm missing something and this shouldn't be needed?

Comment (SteelSlayer) [reply code/modules/antagonists/traitor/contractor/items/contractor_pinpointer.dm:42]: No, I just didn't see the linked issue for pinpointers. This is fine.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 3]
```diff
@@ -652,6 +652,10 @@
 	else
 		..()
 
+/obj/item/card/id/syndicate/Destroy()
+	..()
+	registered_user = null
```
Comment (SteelSlayer) [inline code/game/objects/items/weapons/cards_ids.dm:657]: We always want `return ..()` at the end of destroy
```suggestion
/obj/item/card/id/syndicate/Destroy()
	registered_user = null
	return ..()
```

Comment (DGamerL) [reply code/game/objects/items/weapons/cards_ids.dm]: Sorry my mind isn't fully here, fixed

**Review by SteelSlayer — APPROVED** [review]
